### PR TITLE
Path improvements for use in Pontoon

### PR DESCRIPTION
**Pull request by eemeli**
A collection of fixes to paths required by Pontoon (mozilla/pontoon#3312).

### CODE REVIEW

File: moz/l10n/paths/discover.py [hunk 1]
```diff
@@ -78,9 +58,31 @@ def __init__(
...
+
+        The reference directory has a name matching the `source_locale`
+        and contains files with extensions that appear localizable.
+
+        If `ref_root` is given, the reference directory must be within it.
+        If it does not contain a directory matching `source_locale`
+        but `ref_root` itself contains files with extensions that appear localizable,
+        it is used as the reference directory.
+
+        The default `source_locale` is `['en-US', 'en']`,
```
Comment (mathjazz) [inline moz/l10n/paths/discover.py:83]: What happens with "templates"?

Comment (eemeli) [reply moz/l10n/paths/discover.py:83]: Not treated as anything special. From Pontoon we [call this](https://github.com/mozilla/pontoon/pull/3312/files#diff-b18de80c1f885538e633fd3935679aa1db868c06afc4eecf9af1341df7b707aaR29) with `ref_root` set to the source repo root, which doesn't need to have any special name in order to fall back to using the repo root as the reference root.

Comment (mathjazz) [reply moz/l10n/paths/discover.py:83]: How is it going to detect that https://github.com/mozilla/addons-server/tree/master/locale/templates is the source locale folder, and not https://github.com/mozilla/addons-server/tree/master/locale/en_US?

Comment (eemeli) [reply moz/l10n/paths/discover.py:83]: At least for now, we'll need to call this from Pontoon with `source_locale=["templates", "en-US", "en"]` to keep its behaviour consistent.

**Comment by eemeli** [comment]
I think this batch is complete for now, and I could roll out a new release once this lands. Further iteration may of course be needed later.

File: moz/l10n/paths/discover.py (hunk 1 above)
Comment (mathjazz) [reply moz/l10n/paths/discover.py:83]: Meh, I completely missed `ref_paths` was a function parameter.

**Review by mathjazz — APPROVED** [review]

**Pull request merged**